dlmalloc/src/malloc.c

```diff
@@ -5197,6 +5209,43 @@ static void internal_inspect_all(mstate m,
 }
 #endif /* MALLOC_INSPECT_ALL */
 
+#ifdef __wasilibc_unmodified_upstream // Define a function that initializes the initial state of dlmalloc
+#else
```

Comment:
You could use `#ifndef`.

Reply:
Hi!

Yes, but It seems that current repo has pattern of `#ifdef` {empty} + `#else` as a coding style requirement. E.g. you can find a lot of such patterns in wasi-libc and no patterns with `#ifndef`.

I think that pattern with `#else` in some situations provides more easy way to update from upstream.

@sunfishcode is it true?

Reply:
Yeah, that is what we do elsewhere in wasi-libc. And yes, I sometimes write scripts when working with upstream changes, which are slightly simpler if all the conditionals run the same way. That said, it's not hugely important, so we could change the convention if people find it odd.